### PR TITLE
Move task dispatch events to database from in-memory store.

### LINKED ISSUE #196: Task manager persistence
Task dispatch events shall be persisted in data storage instead of memory (PoC design).  
The persisted events can be used to monitor active tasks, compose update events, and validate task update events.

## Acceptance Criteria

* Events shall be stored in a separate MongoDB collection
* Events shall be removed from the database upon callback

### PR DESCRIPTION
**Pull request by mocsharp**
### Description
Fixes #196  & #252 

- Stores incoming task dispatch events after calling the plug-in
- Upon receiving a task callback event, TM queries the database to look up the stored task dispatch event and constructs a new instance of the specified plug-in. 
- Deletes the task dispatch event from the database upon completion of the callback event.
- Deletes any user accounts created by `ExecuteTask` upon completion of the callback event.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] All tests passed locally.
- [ ] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.
- [x] User guide updated.
- [x] I have updated the changelog

### CODE REVIEW

File: src/TaskManager/TaskManager.cs [hunk 1]
```diff
@@ -348,12 +415,12 @@ private async Task PopulateTemporaryStorageCredentials(params Messaging.Common.S
...
             foreach (var storage in storages)
             {
-                var credentials = await _storageService.CreateTemporaryCredentialsAsync(storage.Bucket, storage.RelativeRootPath, _options.Value.TaskManager.TemporaryStorageCredentialDurationSeconds, _cancellationToken).ConfigureAwait(false);
+                var credeentials = await _storageService.CreateTemporaryCredentialsAsync(storage.Bucket, storage.RelativeRootPath, _options.Value.TaskManager.TemporaryStorageCredentialDurationSeconds, _cancellationToken).ConfigureAwait(false);
                 storage.Credentials = new Credentials
                 {
-                    AccessKey = credentials.AccessKeyId,
-                    AccessToken = credentials.SecretAccessKey,
-                    SessionToken = credentials.SessionToken,
+                    AccessKey = credeentials.AccessKeyId,
```
Comment (jackschofield23) [inline src/TaskManager/TaskManager.cs:421]: spelling mistake

File: src/WorkflowManager/appsettings.Development.json [hunk 2]
```diff
@@ -26,10 +26,10 @@
     "storage": {
       "serviceAssemblyName": "Monai.Deploy.Storage.MinIO.MinIoStorageService, Monai.Deploy.Storage.MinIO",
       "settings": {
-        "endpoint": "localhost:9000",
```
Comment (jackschofield23) [inline src/WorkflowManager/appsettings.Development.json:29]: Can these not be changed, at least for appsettings.development as this is what is configured with the included docker-compose in the repo

Comment (mocsharp) [reply src/WorkflowManager/appsettings.Development.json:29]: Fixed

File: src/WorkflowManager/appsettings.Development.json [hunk 3]
```diff
@@ -39,9 +39,9 @@
     "messaging": {
       "publisherServiceAssemblyName": "Monai.Deploy.Messaging.RabbitMQ.RabbitMQMessagePublisherService, Monai.Deploy.Messaging.RabbitMQ",
       "publisherSettings": {
-        "endpoint": "localhost",
-        "username": "admin",
```
Comment (jackschofield23) [inline src/WorkflowManager/appsettings.Development.json:43]: can you undo these changes for development?

**Review by jackschofield23 — APPROVED** [review]

File: src/WorkflowManager/appsettings.Development.json (hunk 3 above)
Comment (mocsharp) [reply src/WorkflowManager/appsettings.Development.json]: Revert the changes.

**Comment by mocsharp** [comment]
Closing #196 & #252

**Comment by mocsharp** [comment]
Closing #252

**Pull request merged**